chain/index/chain-indexing-overview-for-rpc-providers.MD

```diff
@@ -0,0 +1,218 @@
+# ChainIndexer Documentation for RPC Providers
+  
+## Introduction
+
+This document is for RPC providers and node operators who serve RPC requests walking through the configuration changes, migration flow and operations/maintenance work needed to enable, backfill and maintain the [`ChainIndexer`](#chainindexer-indexing-system).
+
+**Note: If you are a Storage Provider or node operator who does not serve RPC requests (i.e, if `Fevm.EnableEthRPC = false`), you can skip this document as the `ChainIndexer` is already disabled by default**. 
+
+## ChainIndexer Config
+### Enablement
+
+The following must be enabled on an RPC node before starting as they are disabled by default:
+
+```toml
+[ChainIndexer]
+# Enable the ChainIndexer. 
+  EnableIndexer = true 
+
+[Fevm]
+# Enable the ETH RPC APIs.
+  EnableEthRPC = true
+
+[Events]
+# Enable the Actor Events APIs.
+  EnableActorEventsAPI = true
+```
+
+You can learn more about these configuration options and other configuration options available for the `ChainIndexer` [here](https://github.com/filecoin-project/lotus/blob/master/documentation/en/default-lotus-config.toml).
+
+
+### Garbage Collection
+
+The `ChainIndexer` includes a garbage collection (GC) mechanism to manage the amount of historical data retained. By default, GC is disabled to preserve all indexed data.
+
+To configure GC, use the `GCRetentionEpochs` parameter in the `ChainIndexer` section of your config.
+
+The ChainIndexer [periodically runs](https://github.com/filecoin-project/lotus/blob/master/chain/index/gc.go#L15) GC if `GCRetentionEpochs` is > 0 and removes indexed data for epochs older than `(current_head_height - GCRetentionEpochs)`.
+
+```toml
+[ChainIndexer]
+  GCRetentionEpochs = X  # Replace X with your desired value
+```
+
+- Setting `GCRetentionEpochs` to 0 (**default**) disables GC.
+- Any positive value enables GC and determines the number of epochs of historical data to retain.
+
+#### Recommendations
+
+1. **Archival Nodes**: **Keep GC disabled** (`GCRetentionEpochs` = 0) to retain all indexed data.
+
+2. **Non-Archival Nodes**:  Set `GCRetentionEpochs` to match the amount of chain state your node retains 
+(*Example:* if your node is configured to retain 2 days of Filecoin chain state with the Splitstore, set `GCRetentionEpochs` to `retentionDays * epochsPerDay = 2 * 2880 = 5760`).
+**Warning:** Setting this value below the chain state retention period will degrade RPC performance and reliability because the Index will lack data for epochs still present in the chain state.  
+
+### Removed Options
+
+**Note: The following config options no longer exist in Lotus and have been removed in favor of the ChainIndexer config options explained above. They can be removed when upgrading to Lotus v1.31.0.**
+
+```toml
+[Fevm]
+EthTxHashMappingLifetimeDays = 0
```

Comment:
```suggestion
EthTxHashMappingLifetimeDays = 0
  [Events]
  DisableHistoricFilterAPI = false
```

see note in #12421, this one is duplicated with a `moved` redirect

Reply:
Handling in https://github.com/filecoin-project/lotus/pull/12600